akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -816,19 +824,28 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
 
   /**
    * Install a handler on the given inlet that emits received elements on the
-   * given outlet before pulling for more data. `doTerminate` controls whether
+   * given outlet before pulling for more data. `doFinish` and `doFail` control whether
    * completion or failure of the given inlet shall lead to stage termination or not.
-   */
-  final protected def passAlong[Out, In <: Out](from: Inlet[In], to: Outlet[Out], doFinish: Boolean, doFail: Boolean): Unit =
+   * `doPull` instructs to perform one initial pull on the `from` port.
+   */
+  final protected def passAlong[Out, In <: Out](from: Inlet[In], to: Outlet[Out],
+                                                doFinish: Boolean = true, doFail: Boolean = true,
+                                                doPull: Boolean = false): Unit = {
+    val puller = () ⇒ tryPull(from)
```

Comment:
These kinds allocations and the indirection they cause is still giving me the heebie-jeebies